### ISSUE TITLE
response headers

### ISSUE THREAD

**Issue by svnlto** [agraboso/redux-api-middleware#9]
I'm looking for a way to get ahold of the response headers from a `POST` request. How would I go about this?

**Comment by agraboso** [comment]
Currently, you cannot do it. What is your use case?

**Comment by svnlto** [comment]
Need to handle data I'm getting back through 'Authorization' headers.

**Comment by agraboso** [comment]
Are you talking about the `WWW-Authenticate` and `Proxy-Authenticate` headers as per [RFC 7235 section 4.1](https://tools.ietf.org/html/rfc7235#section-4.1)? We could support something very restricted and well-justified like this, but I wouldn't want to clutter `redux-api-middleware` by making it pass all response headers back to the client.

**Comment by svnlto** [comment]
>  but I wouldn't want to clutter redux-api-middleware by making it pass all response headers back to the client.

Well, technically you're doing that already since `fetch` has the headers constructor on the response object you're getting back from your `xhr`.

I do however, see where you're coming from with this. Perhaps adding a new config key which determines which headers are being passed on as part of the payload could be an option?

```js
export function fetchUser(userId, schema = userSchema) {
  return {
    [CALL_API]: {
      types: ['FETCH_USER.REQUEST', 'FETCH_USER.SUCCESS', 'FETCH_USER.FAILURE'],
      endpoint: `/users/${userId}`,
      method: 'GET',
      headers: { credentials: 'same-origin'},
      schema,
      resHeaders: ['Authorization', 'X-SomeToken']
    },
    payload: { somePayload }
  };
}
```

Thoughts?

**Comment by agraboso** [comment]
That sounds good, but I'd need to rearrange a few things. I'll let you know soonish.

**Comment by seokgyo** [comment]
I also need this feature. real-world example in redux also get 'link' data from header.

**Comment by agraboso** [comment]
Solved by `v1.0.0-beta1`.

**Comment by caleb-harrelson** [comment]
For others looking at how to handle custom headers in the response, here's how I did mine.

- Wrap redux-api-middleware with a middleware that adds additional logic to RSAAs.
- That middleware wraps success RSAAs to add a meta property with your handling.
- You can then either return the header in the meta so it's available to all reducers, or do your handling globally in the meta function (as I do below).

Here's a simplified version of my success wrapper, which checks a response header to see if the server wants the client to force a reload (after a deploy) and passes that client-version off to a redux action:
```
const generateSuccessType = (type, dispatch) => {
	return {
		type,
		meta: (action, state, res) => {
			let version = res.headers.get('X-Client-Version')
			if (version) {
				checkClientVersion(version)
			}
		}
	}
}
```
